lib/buffer.js

```diff
@@ -462,8 +462,11 @@ Buffer.prototype.toString = function() {
   } else {
     result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
+  if (result === undefined) {
+    if (this.length > binding.kStringMaxLength)
+      throw new Error('Size exceeds max string length');
```

Comment:
Introducing a new error message is already a breaking change. I'll let others weigh in, but my personal opinion would be to make it a `RangeError`.